### PR TITLE
input, button: various tweaks

### PR DESCRIPTION
**Pull request by austinhallock**
- input
  - added type option
  - added fullWidth option
  - hint colors
- button
  - max-width and ellipsis

### CODE REVIEW

**Comment by austinhallock** [comment]
same changes as other branch, rebase/merge stuff got messy, so it was quickest to just start from clean master

**Comment by austinhallock** [comment]
i also changed base.styl to include nib w/ relative path. if we require zorium-paper/base.styl from our clay-mobile base.styl, for some reason zorium-paper's stylus tries to load in nib.js instead of nib.styl

```
Potentially unhandled rejection [2] ParseError: /home/austin/dev/clay-mobile/node_modules/zorium-paper/node_modules/nib/lib/nib.js:15:108
```

File: base.styl [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@require 'nib'
```
Comment (Zolmeister) [inline base.styl]: This isn't correct. If this really is broken, it needs to be fixed in stylus-loader, not here

**Comment by Zolmeister** [comment]
done

**Comment by austinhallock** [comment]
removed the nib stuff

**Pull request merged**